components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/pip/CarbonAttributeFinder.java

```diff
@@ -136,9 +138,18 @@ public EvaluationResult findAttribute(URI attributeType, URI attributeId, String
 
         List<AttributeValue> attrBag = new ArrayList<AttributeValue>();
         // Get the list of attribute finders who are registered with this particular attribute.
-        List<PIPAttributeFinder> finders = attrFinders.get(attributeId.toString());
 
 
+        List<PIPAttributeFinder> finders = null;
+
+        if (StringUtils.isNotBlank(category.toString())) {
+            finders = attrFinders.get(category.toString());
+        }
+
+        if (CollectionUtils.isEmpty(finders)) {
+            finders = attrFinders.get(attributeId.toString());
+        }
+
         if (finders == null || finders.size() == 0) {
             finders = attrFinders.get(attributeId.toString());
```

Comment:
Why are we trying to get attributeFinders based on attributeId once again? 
Aren't we doing that in line 150?